### PR TITLE
CPP: Add syntax examples to QLDoc in various files

### PR DESCRIPTION
**Pull request by geoffw0**
A number of straightforward cases here.

For https://jira.semmle.com/browse/CPP-407.

### CODE REVIEW

File: cpp/ql/src/semmle/code/cpp/Comments.qll [hunk 1]
```diff
@@ -2,7 +2,11 @@ import semmle.code.cpp.Location
 import semmle.code.cpp.Element
 
 /**
- * A C/C++ comment.
+ * A C/C++ comment. For example the comment in the following code:
+ * ```
+ * // C++ style single-line comment
+ * ```
+ * or a C style comment (which starts with `/*`).
  */
```
Comment (zlaski-semmle) [inline cpp/ql/src/semmle/code/cpp/Comments.qll]: I believe that C99 also allows single-line comments. Also, why not just say
>  * `/* C style comment */`
right after
 > * `// C++ style single-line comment`
?

Comment (geoffw0) [reply cpp/ql/src/semmle/code/cpp/Comments.qll]: Yes, modern C allows C++-style comments, but they're still understood to be C++ *style*.

Regarding `/* C style comment */`, that's what I wanted to write but it is sadly impossible to put that in the QLDoc comment cleanly without the `*/` closing the QLDoc comment itself and causing problems.  Hence the irregularity.

Comment (zlaski-semmle) [reply cpp/ql/src/semmle/code/cpp/Comments.qll]: Have you tried escaping the asterisk, i.e., `\*/` ?  Markdown languages appear to support it.

Comment (geoffw0) [reply cpp/ql/src/semmle/code/cpp/Comments.qll]: `*&#47;` has been suggested, the trouble is that I suspect QLDoc is read plaintext as frequently as it is read in rendered form.  After all a `/**` QLDoc comment is also a `/*` code comment.  So I'm reluctant to add strange encodings that aren't readable in plaintext form (`\*/` would have been better but it doesn't seem to work).

**Pull request merged**